tasks/growing-commandlog/src/main/java/voltdb/tasks/GrowingCommandlog.java

```diff
@@ -0,0 +1,189 @@
+
+    private TaskHelper helper;
+    private int maxSegmentCount;
+
+    public static String validateParameters(TaskHelper helper, int maxSegmentCount) {
+        CompoundErrors errors = new CompoundErrors();
+        if (maxSegmentCount <= 0) {
+            errors.addErrorMessage(MSG_ERROR_BAD_SEGMENT_CNT);
+        }
+        // Add more errors or warnings if there are more parameters. Other possible parameters
+        // could be how old the last truncation snapshot is allowed to be or how many times
+        // the warning will be promoted to an error or even a call to @Shutdown
+        return errors.getErrorMessage();
+    }
+
+    public void initialize(TaskHelper helper, int maxSegmentCount) {
+        this.helper = helper;
+        this.maxSegmentCount = maxSegmentCount <= 0 ? Integer.MAX_VALUE : maxSegmentCount;
+    }
+
+    @Override
+    public Action getFirstAction() {
+        return Action.procedureCall(this::handleStatisticsResult, "@Statistics", "COMMANDLOG", 0);
+    }
+
+    @Override
+    public boolean isReadOnly() {
+        return true;
+    }
+
+    @Override
+    public Collection<String> getDependencies() {
+        // This class has no external dependencies outside of java standard library and voltdb
+        return Collections.emptyList();
+    }
+
+    /**
+     * Callback to handle the result of the statistics query and return the next statistics call
+     *
+     * @param result of statistics query
+     * @return next statistics action
+     */
+    private Action handleStatisticsResult(ActionResult result) {
+        ClientResponse response = result.getResponse();
+        ProblemMsg msg = null;
+
+        if (response.getStatus() == ClientResponse.SUCCESS) {
+            VoltTable stats = response.getResults()[0];
+
+            while (stats.advanceRow()) {
+                int currSegmentCnt = (int)stats.getLong("IN_USE_SEGMENT_COUNT");
+                if (currSegmentCnt > maxSegmentCount) {
+                    if (msg == null) {
+                        msg = new ProblemMsg();
+                    }
+                    String hostname = stats.getString("HOSTNAME");
+                    msg.udpate(currSegmentCnt, hostname);
+                }
+            }
+            if (msg != null) {
+                helper.logWarning(msg.buildWarningString() + " The expected maximum is " + maxSegmentCount +
+                        " segments. This is an indication that Truncation Snapshots are no longer completing" +
+                        " either because there is a problem in the snapshot subsystem or because other types" +
+                        " of snapshots are in progress. Large CommandLogs will slow down a cluster recovery. If" +
+                        " truncation snapshots are no longer completing, it is recommended that the cluster be" +
+                        " shutdown and recovered as soon as possible.");
+            }
+        }
```

Comment:
Maybe there should be an else block which logs the failure. This could log every N failures so it can be detected if the task is completely failing